src/performance/metrics/PerformanceUtil.cpp

```diff
@@ -0,0 +1,1369 @@
+        read(sockfd, data, 300);
+        response = (data);
+        response = utils.trim_copy(response, " \f\n\r\t\v");
+        scheduler_logger.log("Received : " + JasmineGraphInstanceProtocol::OK, "info");
+
+        if (response.compare(JasmineGraphInstanceProtocol::HOST_OK) == 0) {
+            write(sockfd, JasmineGraphInstanceProtocol::PERFORMANCE_STATISTICS.c_str(),
+                  JasmineGraphInstanceProtocol::PERFORMANCE_STATISTICS.size());
+            scheduler_logger.log("Sent : " + JasmineGraphInstanceProtocol::PERFORMANCE_STATISTICS, "info");
+            bzero(data, 301);
+            read(sockfd, data, 300);
+            response = (data);
+            response = utils.trim_copy(response, " \f\n\r\t\v");
+            scheduler_logger.log("Received : " + response, "info");
+
+            if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
+                write(sockfd, isVMStatManager.c_str(), isVMStatManager.size());
+                scheduler_logger.log("Sent : VM Manager Status " + isVMStatManager, "info");
+
+                bzero(data, 301);
+                read(sockfd, data, 300);
+                response = (data);
+                response = utils.trim_copy(response, " \f\n\r\t\v");
+
+                if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
+                    write(sockfd, isResourceAllocationRequired.c_str(), isResourceAllocationRequired.size());
+                    scheduler_logger.log("Sent : Resource Allocation Requested " + isResourceAllocationRequired, "info");
+
+                    bzero(data, 301);
+                    read(sockfd, data, 300);
+                    response = (data);
+                    response = utils.trim_copy(response, " \f\n\r\t\v");
+
+                    scheduler_logger.log("Performance Response " + response, "info");
+
+                    std::vector<std::string> strArr = Utils::split(response, ',');
+
+                    std::string processTime = strArr[0];
+                    std::string memoryUsage = strArr[1];
+                    std::string cpuUsage = strArr[2];
+                    std::string loadAverage = strArr[3];
+
+                    ResourceUsageInfo resourceUsageInfo;
+                    resourceUsageInfo.elapsedTime = std::to_string(elapsedTime);
+                    resourceUsageInfo.loadAverage = loadAverage;
+                    resourceUsageInfo.memoryUsage = memoryUsage;
+
+
+                    if (!resourceUsageMap[placeId].empty()) {
+                        resourceUsageMap[placeId].push_back(resourceUsageInfo);
+                    } else {
+                        std::vector<ResourceUsageInfo> resourceUsageVector;
+
+                        resourceUsageVector.push_back(resourceUsageInfo);
+
+                        resourceUsageMap[placeId] = resourceUsageVector;
+                    }
+
+                    /*if (isVMStatManager == "true" && strArr.size() > 3 && isResourceAllocationRequired == "true") {
```

Comment:
Why do we have this segment commented?

Reply:
Removed the commented section from the code

Reply:
Ok